File: 0400-0499/0412-fizz-buzz/0412-fizz-buzz.py
"""
One of the most stupid questions. Perhaps the only trick is to keep 2 running counters and reset them every 3 and 5
iterations, which is slightly faster than dividing the numbers.
"""
class Solution(object):
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        res = []

        fizz, buzz = 0, 0
        for i in range(1, n+1):
            fizz += 1
            buzz += 1
            word = ""

            if fizz == 3:
                word += "Fizz"
                fizz = 0
            if buzz == 5:
                word += "Buzz"
                buzz = 0
            if len(word) == 0:
                word = str(i)

            res.append(word)

        return res

File: 0400-0499/0412-fizz-buzz/0412-fizz-buzz.cpp
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
	vector<string> fizzBuzz(int n) {
		vector<string> ret;
		
		for (int i = 1, j = 1, k = 1; i <= n; i++, j++, k++) {
			if (j == 3 && k == 5) {
				j = 0, k = 0;
				ret.push_back("FizzBuzz");
			} else if (j == 3) {
				j = 0;
				ret.push_back("Fizz");
			} else if (k == 5) {
				k = 0;
				ret.push_back("Buzz");
			} else {
				ret.push_back(to_string(i));
			}
		}
		
		return ret;
	}
};